File: app/lib/models/category.dart
class Category {
  final int id;
  final String name;
  final String description;
  final int sortOrder;

  Category({
    required this.id,
    required this.name,
    required this.description,
    required this.sortOrder,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      sortOrder: json['sort_order'] ?? json['sortOrder'] ?? 0,
    );
  }
}

File: backend/models/category.go
package models

type Category struct {
	ID          uint   `gorm:"primarykey" json:"id"`
	Name        string `gorm:"size:50;not null;unique" json:"name"`
	Description string `gorm:"size:200" json:"description"`
	SortOrder   int    `gorm:"default:0" json:"sort_order"`
}
